### ISSUE TITLE
No easy way to created concatenated declaration files

### ISSUE THREAD

**Issue by keesey** [ivogabe/gulp-typescript#77]
I have a project with many `.ts` files. For distribution I compile them into a single `.d.ts` file and a single `.js` file. The `.js` file works great, but the `.d.ts` file has all the references to the source TypeScript files' declaration files, even though they don't exist in the distribution, causing the concatenated `.d.ts` file to be useless.

Simplified `gulpfile.js`:

```
var concat = require('gulp-concat');
var gulp = require('gulp');
var merge = require('merge2');
var typescript = require('gulp-typescript');

var TYPESCRIPT_PROJECT = typescript.createProject({
  declarationFiles: true,
  sortOutput: true
});

gulp.task('scripts', function() {
  var tsResult = gulp
    .src('./app/scripts/**/*.ts')
    .pipe(typescript(TYPESCRIPT_PROJECT));
  return merge(
    tsResult.dts
      .pipe(concat('myPackage.d.ts'))
      .pipe(gulp.dest('./dist/scripts')),
    tsResult.js
      .pipe(concat('myPackage.js'))
      .pipe(gulp.dest('./dist/scripts'))
  );
});
```

Simplified source TS files:

`app/scripts/a.ts`
```
/// <reference path="../bower_components/dt-jquery/jquery.d.ts" />

var a = 1;
```

`app/scripts/b.ts`
```
/// <reference path="a.ts" />

var b = a + 1;
```

Desired `myPackage.d.ts` output:
```
/// <reference path="../bower_components/dt-jquery/jquery.d.ts" />

declare var a: number;

declare var b: number;
```

Actual `myPackage.d.ts` output:
```
/// <reference path="../bower_components/dt-jquery/jquery.d.ts" />

declare var a: number;

/// <reference path="a.d.ts" />

declare var b: number;
```

I need a way to exclude the references that are generated as part of the active `gulp-typescript` process (e.g., `a.d.ts` in the example), but not the external, preexisting references (e.g., `jquery.d.ts` in the example).

Currently I'm using `gulp-replace` as a workaround, but it's a hack, and the size of the project is starting to cause `gulp-replace` to fail.

**Comment by keesey** [comment]
Put another way, I need to include all `/// <reference path="****.d.ts" />` comments and exclude all `/// <reference path="****.ts" />` comments (which currently compile to `/// <reference path="****.d.ts" />` comments, making them distinguish after compiling).

**Comment by ivogabe** [comment]
I've implemented the `out` option for this in #80. Will merge it into the master soon, I need to do some testing first.

**Comment by keesey** [comment]
Great!
